### PR TITLE
Catch and log errors originating from callbacks in scripts

### LINKED ISSUE #160: Adapter crashes if script has an error that is not immediately invoked from user code
Example:
https://forum.iobroker.net/viewtopic.php?f=22&t=15700&p=165320#p165320

### PR DESCRIPTION
**Pull request by AlCalzone**
Probably fixes #160

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -1191,3 +1191,18 @@ function getData(callback) {
...
+// Catch unhandled errors originating from callbacks in scripts
+// These are not caught by wrapping the execution code in try-catch
+process.on("uncaughtException", e => {
+    const scriptCodeMarker = "script.js.";
+    if (typeof e.stack === "string" && e.stack.indexOf(scriptCodeMarker) > -1) {
+        // This is a script error
+        let scriptName = e.stack.substr(e.stack.indexOf(scriptCodeMarker));
+        scriptName = scriptName.substr(0, scriptName.indexOf(":"));
+        context.logError(scriptName, e);
+        // TODO: should we stop the script execution now?
```
Comment (AlCalzone) [inline main.js]: @GermanBluefox Should we terminate the script here or keep it running?

Comment (BuZZy1337) [reply main.js]: Maybe we could implement an yellow "!" instead of the green play and red pause button to show that there is a problem with the script?

Comment (GermanBluefox) [reply main.js]: Keep it running. May be it is heating...
And orange play button Is interesting

**Comment by AlCalzone** [comment]
Okay, then I merge as soon as I have confirmation this works.
I'll add the orange/yellow button as an issue for the future.

**Comment by AlCalzone** [comment]
@GermanBluefox Apparently this does not work because there's another handler for `process.on("unhandledException")` somewhere else which causes the js-controller to log an error and restart the adapter. Can you point me to where the automatic restart on error happens? 

I'd like to add a "handled" property on errors caught here, so the JS-adapter is not restarted when it has handled the error itself.

**Comment by AlCalzone** [comment]
Nevermind, I found a better solution. Will update this PR soon.

**Comment by AlCalzone** [comment]
I changed the handling code into a method that will be called from the js-controller. This requires https://github.com/ioBroker/ioBroker.js-controller/pull/234 to be merged.

As a result, when an unhandled script error occurs, the `error` method will be called from the js-controller instead of immediately restarting the script adapter. When the adapter tells the js-controller that it has handled the error itself (by returning `true`), no restart will be performed.

**Pull request merged**